oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2AuthorizationCodeAuthenticationProvider.java

```diff
@@ -149,6 +149,12 @@ public Authentication authenticate(Authentication authentication) throws Authent
 		}
 
 		if (!authorizationCode.isActive()) {
+			if (authorizationCode.isInvalidated()) {
+				invalidateAuthorizationTokens(authorization);
```

Comment:
Please add a test for this case.

Reply:
I added logic to an existing test case to verify whether the access and refresh tokens have been invalidated.
(authenticateWhenInvalidatedCodeThenThrowOAuth2AuthenticationException)